src/window.rs

```diff
@@ -56,6 +56,48 @@ struct WindowWrapper {
     fullscreen: bool,
 }
 
+pub fn window_geometry() -> Result<(u64, u64), String> {
+    let prefix = "--geometry=";
+
+    std::env::args()
+        .filter(|arg| arg.starts_with(prefix))
+        .next()
+        .map_or(Ok(INITIAL_DIMENSIONS), |arg| {
+            let input = &arg[prefix.len()..];
+            let invalid_parse_err = format!(
+                "Invalid geometry: {}\nValid format: <width>x<height>",
+                input
+            );
+
+            input
+                .split('x')
+                .map(|dimension| {
+                    dimension
+                        .parse::<u64>()
+                        .or(Err(invalid_parse_err.as_str()))
+                        .and_then(|dimension| {
+                            if dimension > 0 {
+                                Ok(dimension)
+                            } else {
+                                Err("Invalid geometry: Window dimensions should be greater than 0.")
+                            }
+                        })
+                })
+                .collect::<Result<Vec<_>, &str>>()
+                .and_then(|dimensions| {
```

Comment:
welp :l I missed that we needed a tuple instead of a vec was a thing, so my proposal isn't much better really. Whatever. I'll just merge it and think about it some more to see if theres something to be done.